### PR TITLE
feat: previous, next buttons on docs

### PR DESCRIPTION
**Pull request by jeremybosma**
# Changes/Additions being made

Previous and next buttons for components on docs, PostLayout and ComponentLayout

## Rationale behind the change or addition

On mobile navigation was a little annoying

### Quality check

- [ ] Are there any erroneous console logs, debuggers or leftover code in your changes?

### CODE REVIEW

File: src/components/NavigationButtons.astro [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+---
+
+<Spread>
+  {
+    prevPage && (
+      <Button
+        href={prevPage.href}
+        variant="secondary"
+      >
+        ← {prevPage.title}
```
Comment (silveltman) [inline src/components/NavigationButtons.astro:51]: Do the following instead:
```
  <Button
      href={prevPage?.href}
      text={prevPage?.title}
      variant="secondary"
    >
    ```
    
    This way you do not need the wrapping check, bacause the button will only render if it has a text. That is built into the component

File: src/components/NavigationButtons.astro [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      </Button>
+    )
+  }
+  {
+    nextPage && (
```
Comment (silveltman) [inline src/components/NavigationButtons.astro:56]: what if only the next button is present, will it be at the left side?

If so, add marginleft auto to it

File: src/components/NavigationButtons.astro [hunk 3]
```diff
@@ -0,0 +1,66 @@
+---
```
Comment (silveltman) [inline src/components/NavigationButtons.astro:1]: Place the component in the components/docs folder, as it is only used in here

File: src/layouts/ComponentLayout.astro [hunk 4]
```diff
@@ -18,6 +21,7 @@ const { Content } = await render()
     <Prose>
       <Content />
     </Prose>
+    <NavigationButtons currentSlug={currentSlug} />
```
Comment (silveltman) [inline src/layouts/ComponentLayout.astro:24]: what if there is a normal post where we do not want the naviagationbuttons, are they still visible or not?

**Review by silveltman — CHANGES_REQUESTED** [review]
also, build error. See netlify logs

File: src/layouts/ComponentLayout.astro (hunk 4 above)
Comment (jeremybosma) [reply src/layouts/ComponentLayout.astro:24]: In the postlayout? Depending on if its defined in the sidebar but for example on a blog-like page not defined in the sidebar settings, no.

**Comment by jeremybosma** [comment]
It builds locally and on cloudfare, on netlify its giving build errors from code that already has been removed and does not exist in the codebase anymore.

**Review by silveltman — DISMISSED** [review]

**Review by silveltman — APPROVED** [review]

**Pull request merged**